### PR TITLE
Improve docs for torch and torch.Tensor

### PR DESCRIPTION
**Pull request by ssnl**
Makes the documents overall much more consistent. Changes include:

1. raw docstrings.
2. improves / add some math expressions
3. makes doc for certain methods clearer, e.g. `chunk` and `split`
4. use a consistent format for tensor shape
5. [Tensor, `Tensor`, `tensor`, etc.] -> tensor when not referring the the class
6. fixes several formatting errors caused by newline

### CODE REVIEW

File: torch/functional.py [hunk 1]
```diff
@@ -9,15 +9,15 @@
 
 
 def split(tensor, split_size, dim=0):
-    """Splits the tensor into equally sized chunks (if possible).
+    """Splits the tensor into chunks all of a certain size (if possible).
```
Comment (colesbury) [inline torch/functional.py]: This seems a bit awkwardly phrased.

Maybe, "Splits the tensor into sub-tensors of size :attr:`split_size`."

**Review by colesbury — COMMENTED** [review]
Nice!

**Comment by colesbury** [comment]
@SsnL can please fix the lint errors?

https://travis-ci.org/pytorch/pytorch/jobs/310181860

**Comment by SsnL** [comment]
Sorry about those. Fixing right now!

**Comment by SsnL** [comment]
For some of those.. if I break the line, sphinx will generate weird format. I tried to look for other workarounds but didn't find any. I'll do some more search.

**Pull request merged**